src/textarea/textarea.tsx

```diff
@@ -9,25 +9,127 @@ interface Props {
     className?: string;
     placeholder?: string;
     disabled?: boolean;
+    isCharacterLimitEnabled?: boolean;
 }
+interface State {
+    expanded: boolean;
+    rows: number;
+    minRows: number;
+    maxRows: number;
+    value: string;
+    limit: number;
+    characterLimit: number;
+    stroke: string;
+    strokeDasharray: string;
+    outputText: string;
+    radius: number;
+}
+
+class Textarea extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            expanded: false,
+            rows: this.props.rows ? this.props.rows : 3,
+            minRows: 3,
+            maxRows: 10,
+            value: '',
+            limit: 255,
+            characterLimit: 255,
+            stroke: '',
+            strokeDasharray: '',
+            outputText: '',
+            radius: 11,
+        };
+        this.onChange = this.onChange.bind(this);
+    }
+    public render() {
+        return (
+            <div className="textarea-component">
+                <textarea
+                    id={this.props.id}
+                    className={`form-control text-design ${this.state.stroke} ${
+                        this.props.isCharacterLimitEnabled ? 'character-limited' : ''
+                    } ${this.props.className || ''}`}
+                    rows={this.state.rows}
+                    onBlur={this.props.onBlur}
+                    onChange={this.onChange}
+                    placeholder={this.props.placeholder}
+                    data-qa={`textarea--element-${this.props.id}`}
+                    value={this.props.value}
+                    disabled={this.props.disabled}
+                />
+                {this.props.isCharacterLimitEnabled && (
+                    <svg>
+                        <circle id="gray" cx="50%" cy="50%" r={this.state.radius} />
```

Comment:
These `50%`s seem extractable, also for the following usages

Reply:
done